test_conformance/basic/test_sizeof.c

```diff
@@ -49,8 +49,15 @@ cl_int get_type_size( cl_context context, cl_command_queue queue, const char *ty
     {
         sizeof_kernel_code[0] = "#pragma OPENCL EXTENSION cl_khr_fp16 : enable\n";
     }
+    size_t major = 0;
+    size_t minor = 0;
+    bool deviceLt20 = false;
+    int error = get_device_version(device, &major, &minor);
+    if (major < 2) {
+        deviceLt20 = true;
+    }
 
-    cl_int err = create_single_kernel_helper_with_build_options(context, &p, &k, 4, sizeof_kernel_code, "test_sizeof", "-cl-std=CL2.0");
+    cl_int err = create_single_kernel_helper_with_build_options(context, &p, &k, 4, sizeof_kernel_code, "test_sizeof", deviceLt20 ? "" : "-cl-std=CL2.0");
```

Comment:
As far as I can tell with a quick read of the code, the only thing that requires CL2 in this test is the `void*` pointer case. The compatibility suite has got `global void*` (which was needed for one implementation). I'm fine with what you've done but we'll need to make the creation of the table of pointer types conditional to the version when folding the basic compatibility suite into the main basic suite.